File: src/algebra/GCD.java
package algebra;


public class GCD {

    static int gcd(int a, int b) {
        return b == 0 ? a : gcd(b, a % b);
    }

    static int lcm(int a, int b) {
        return (a/gcd(a,b)) * b;
    }

    public static void main(String[] args) {
        System.out.println(gcd(-10, -21));
        System.out.println(lcm(10, 15));
    }
}

File: src/main/kotlin/algebra/GCD.kt
package algebra

fun gcd(a: Int, b: Int) : Int {
    return if (b == 0) a else gcd(b, a % b)
}

fun lcd(a: Int, b: Int): Int {
    return a / gcd(a, b) * b
}

fun main(args : Array<String>) {
    println(gcd(21, 41))
    println(lcd(3, 7))
}